operations/mimir-mixin/dashboards/dashboard-utils.libsonnet

```
@@ -495,22 +495,56 @@ local utils = import 'mixin-utils/utils.libsonnet';
       $.panel(title) +
       $.queryPanel(
         [
-          'kube_horizontalpodautoscaler_spec_min_replicas{%s, horizontalpodautoscaler=~"%s"}' % [$.namespaceMatcher(), $._config.autoscaling[component].hpa_name],
-          'kube_horizontalpodautoscaler_spec_max_replicas{%s, horizontalpodautoscaler=~"%s"}' % [$.namespaceMatcher(), $._config.autoscaling[component].hpa_name],
-          'kube_horizontalpodautoscaler_status_current_replicas{%s, horizontalpodautoscaler=~"%s"}' % [$.namespaceMatcher(), $._config.autoscaling[component].hpa_name],
+          |||
+            kube_horizontalpodautoscaler_spec_max_replicas{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+            # Add the scaletargetref_name label for readability
+            + on (%(cluster_labels)s, horizontalpodautoscaler) group_left (scaletargetref_name)
+              0*kube_horizontalpodautoscaler_info{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+          ||| % {
+            namespace_matcher: $.namespaceMatcher(),
+            hpa_name: $._config.autoscaling[component].hpa_name,
+            cluster_labels: std.join(', ', $._config.cluster_labels),
+          },
+          |||
+            kube_horizontalpodautoscaler_status_current_replicas{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+            # HPA doesn't go to 0 replicas, so we multiply by 0 if the HPA is not active
+            * on (%(cluster_labels)s, horizontalpodautoscaler)
+              kube_horizontalpodautoscaler_status_condition{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s", condition="ScalingActive", status="true"}
+            # Add the scaletargetref_name label for readability
+            + on (%(cluster_labels)s, horizontalpodautoscaler) group_left (scaletargetref_name)
+              0*kube_horizontalpodautoscaler_info{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+          ||| % {
+            namespace_matcher: $.namespaceMatcher(),
+            hpa_name: $._config.autoscaling[component].hpa_name,
+            cluster_labels: std.join(', ', $._config.cluster_labels),
+          },
         ],
         [
-          'Min',
-          'Max',
-          'Current',
+          'Max {{ scaletargetref_name }}',
+          'Current {{ scaletargetref_name }}',
         ],
       ) +
       $.panelDescription(
         title,
         |||
-          The minimum, maximum, and current number of %s replicas.
+          The maximum and current number of %s replicas.
+          Please note that the current number of replicas can still show 1 replica even when scaled to 0.
+          Since HPA never reports 0 replicas, the query will report 0 only if the HPA is not active.
```

```suggestion
          Because HPA never reports 0 replicas, the query will report 0 only if the HPA is not active.
```
For rationale, see "since" at https://developers.google.com/style/word-list#letter-s.